packages/sites/src/drafts/upgrade-draft-schema.ts

```diff
@@ -0,0 +1,27 @@
+import { deepSet, getProp, IDraft } from "@esri/hub-common";
+import { SITE_SCHEMA_VERSION } from "../site-schema-version";
+import { _ensureTelemetry } from "../_ensure-telemetry";
+
+const schemaVersionPath = "item.properties.schemaVersion";
+const initialDraftVersion = 1.3;
+
+/**
+ * Applies the schema upgrades
+ * @param draft IDraft
+ */
+export function upgradeDraftSchema(draft: IDraft) {
+  if (getProp(draft, "item.properties.schemaVersion") === undefined) {
+    deepSet(draft, schemaVersionPath, initialDraftVersion);
+  }
+
+  if (getProp(draft, "item.properties.schemaVersion") === SITE_SCHEMA_VERSION) {
+    return draft;
+  } else {
+    let migrated = draft;
+    // apply site schema upgrade functions in order...
+    // don't have do do them all since drafts only got released
+    // at version 1.3
+    migrated = _ensureTelemetry<IDraft>(draft);
```

Comment:
Minor, `<IDraft>` & `<IModel>` aren't necessary before invoking `_ensureTelemetry`.

Reply:
Hi @rweber-esri . I changed `_ensureTelemetry` to accept a Typescript generic [here](https://github.com/Esri/hub.js/pull/500/files#diff-eff07968780063bba314ca024cbb534e5ebd32408579a368817af36a484bfb42R16). Could you clarify what you mean?

Reply:
FWIW, this isn't _super_ important, just pointing out we could save a few bytes is all. The `T` in that generic signature represents whatever type is passed to the function (in this case either IModel or IDraft). Since the signature defines that the method will receive an object of type `T` & return an object of type `T`, it means the type can be inferred at runtime and we don't need to explicitly tell `_ensureTelemetry` whether we're passing an `IDraft` or `IModel`. E.g.

```
// migrated = _ensureTelemetry<IDraft>(draft); // can be reduced to just
migrated = _ensureTelemetry(draft);

// and

// model = _ensureTelemetry<IModel>(model); // can be reduced to just
model = _ensureTelemetry(model);
```

That being said, there's nothing _wrong_ w/ what you did, could just reduce it a bit is all.